be1-go/message/messagedata/coin_post_transaction.go

```diff
@@ -113,6 +129,50 @@ func (message PostTransaction) Verify() error {
 	return nil
 }
 
+func (message PostTransaction) verifySignature() error {
+	var sigComp []string
+
+	for _, inp := range message.Transaction.Inputs {
+		hash := inp.Hash
+		sigComp = append(sigComp, hash)
+
+		index := strconv.Itoa(inp.Index)
+		sigComp = append(sigComp, index)
+	}
+
+	for _, out := range message.Transaction.Outputs {
+		value := strconv.Itoa(out.Value)
+		sigComp = append(sigComp, value)
+
+		typee := out.Script.Type
+		sigComp = append(sigComp, typee)
+
+		pubKey := out.Script.PubKeyHash
+		sigComp = append(sigComp, pubKey)
+	}
+
+	dataBytes := []byte(strings.Join(sigComp, ""))
+
+	for _, inp := range message.Transaction.Inputs {
+		signatureBytes, err := base64.URLEncoding.DecodeString(inp.Script.Sig)
+		if err != nil {
+			return xerrors.Errorf("failed to decode signature string: %v", err)
+		}
+
+		publicKeySender, err := base64.URLEncoding.DecodeString(inp.Script.PubKey)
+		if err != nil {
+			return xerrors.Errorf("failed to decode public key string: %v", err)
+		}
+
+		err = schnorr.VerifyWithChecks(crypto.Suite, publicKeySender, dataBytes, signatureBytes)
+		if err != nil {
+			return xerrors.Errorf("failed to verify signature : %v", err)
```

Comment:
I think that a custom error from `answer` should be applied here instead of `xerrors`

Reply:
Only Here ? or for all of them ? What would concretely differ from doing that?